app/components/date-time-modal/date-time-modal-controller.js

```diff
@@ -20,34 +34,47 @@ angular.module('bulbs.cms.dateTimeModal.controller', [
         // default to now if no time given
         $scope.tempDatetime = moment();
       }
-
-      $scope.TIMEZONE_LABEL = moment.tz(CmsConfig.getTimezoneName()).format('z');
+      $scope.tempTime = angular.copy($scope.tempDatetime);
 
       var timeNowWithOffset = function () {
         return moment.tz(CmsConfig.getTimezoneName());
       };
 
-      // callback function for using datetime calendar because it doesn't work at all in a sensible way
+      // callback function for using datetime calendar because it doesn't work
+      //  at all in a sensible way
       $scope.setDate = function (newDate) {
-        $scope.tempDatetime = moment(newDate);
+        var newDateAsMoment = moment(newDate);
+        $scope.tempDatetime = ($scope.tempDatetime || moment())
+          .year(newDateAsMoment.year())
+          .month(newDateAsMoment.month())
+          .date(newDateAsMoment.date());
       };
 
       $scope.setDateToday = function () {
         var now = timeNowWithOffset();
-        $scope.tempDatetime = moment().year(now.year()).month(now.month()).date(now.date());
+        $scope.tempDatetime = ($scope.tempDatetime ||  moment())
```

Comment:
I feel like `setDate`, `setDateToday`, and `setDateTomorrow` read so similarly, that maybe they should be combined into one function?